node/rate_limiter.js

```diff
@@ -124,10 +124,35 @@ function removeServiceCounter(serviceName) {
     delete self.counters[serviceName];
 };
 
+RateLimiter.prototype.updateExemptServices =
+function updateExemptServices(exemptServices) {
+    var self = this;
+    self.exemptServices = exemptServices;
+};
+
+RateLimiter.prototype.updateRpsLimitForServiceName =
+function updateRpsLimitForServiceName(rpsLimitForServiceName) {
+    var self = this;
+    // do a complete update
+    var keys = Object.keys(self.rpsLimitForServiceName);
+    for (var i = 0; i < keys.length; i++) {
+        var name = keys[i];
+        self.updateServiceLimit(name, rpsLimitForServiceName[name]);
+    }
+};
+
 RateLimiter.prototype.updateServiceLimit =
 function updateServiceLimit(serviceName, limit) {
     var self = this;
-    self.rpsLimitForServiceName[serviceName] = limit;
+
+    if (limit === undefined) {
```

Comment:
it's cleaner to check for `if (limit === null)` and then have `updateServiceLimit('foo', null)`